src/gnc/attitude_estimator.cpp

```diff
@@ -438,10 +439,35 @@ void attitude_estimator_reset(AttitudeEstimatorState &state,
     double t, lin::Vector3d const &r_ecef, lin::Vector3f const &b_body,
     lin::Vector3f const &s_body) {
   GNC_ASSERT_NORMALIZED(s_body);
-
-  // TODO : Implement this
-  // https://github.com/pathfinder-for-autonomous-navigation/psim/issues/187
+  
   lin::Vector4f q_body_eci = lin::nans<float, 4, 1>();
+  
+  lin::Vector3f b_eci{};
+  lin::Vector4f q{};
+  env::magnetic_field(t, lin::Vector3f(r_ecef), b_eci);  // b_eci = b_ecef
+  env::earth_attitude(t, q);                             // q = q_ecef_eci
+  utl::quat_conj(q);                                     // q = q_eci_ecef
+  utl::rotate_frame(q, b_eci);                           // b_eci = b_eci  
```

Comment:
I think the comment should say something like b_eci = b_ecef? unless I'm misunderstanding.

Reply:
Here, `b_eci` was previsouly equal to `b_ecef` (see line 447) and `q` is `q_eci_ecef`. Therefore the transformation moves the magnetic field reading from ECEF to EC.